swaybar/bar.c

```diff
@@ -50,18 +57,39 @@ static void spawn_status_cmd_proc(struct bar *bar) {
 	}
 }
 
+#ifdef ENABLE_TRAY
+static void spawn_xembed_sni_proxy() {
+	pid_t pid = fork();
+	if (pid == 0) {
+		int wstatus;
+		do {
+			pid = fork();
+			if (pid == 0) {
+				execlp("xembedsniproxy", "xembedsniproxy", NULL);
+				_exit(EXIT_FAILURE);
+			}
+			waitpid(pid, &wstatus, 0);
+		} while (!WIFEXITED(wstatus));
+		_exit(EXIT_FAILURE);
+	}
+}
+#endif
```

Comment:
Consider moving this as well, just have a tray upkeep function that checks this and refires it if not running